### PR TITLE
Atom bamboo integration

### PR DESCRIPTION
**Pull request by samadejacobs**
This PR adds ATOM model to bamboo integration test. 
Only Lassen is supported at this time.

Bamboo is green!

### CODE REVIEW

**Review by timmoon10 — CHANGES_REQUESTED** [review]
I think the ATOM application stuff is fine and it can be merged in immediately as a separate PR. The Bamboo stuff will take some work though.
- I believe the Bamboo build is still using CUDA 10.1 and cuDNN 7.6. The GRU layer requires CUDA 11.0 and cuDNN 8.0.4.
- If the dataset is large, would it be better to run this test weekly rather than nightly? If so, we can copy some code from AlexNet:
https://github.com/LLNL/lbann/blob/57116ecc030c0d17bc941f81131c1a335bc2c4ad/bamboo/integration_tests/test_integration_alexnet.py#L136 We can also adapt this to only run on Lassen. I see in your Bamboo that the test is failing on all non-Lassen systems (an alternative is to use PR #1685).

**Comment by samadejacobs** [comment]
Made two additional commit (1) skip other systems but Lassen (2) skip daily for weekly

**Review by bvanessen — CHANGES_REQUESTED** [review]
@samadejacobs Here is the question.

File: applications/ATOM/train_atom_wae.py [hunk 1]
```diff
@@ -59,9 +59,9 @@ def construct_lc_launcher_args():
     parser.add_argument("--dump-outputs-dir", type=str, default=None)
     parser.add_argument("--dump-outputs-interval", type=int, default=10)
     parser.add_argument("--dump-model-dir", type=str, default=None)
-    parser.add_argument("--num_samples", type=int, default=None)
-    parser.add_argument("--num_train_samples", type=int, default=None)
-    parser.add_argument("--num_test_samples", type=int, default=None)
+    parser.add_argument("--num_samples", type=int, default=10000)
```
[the remark this answers is not in the document]
Comment (samadejacobs) [reply applications/ATOM/train_atom_wae.py]: I have temporarily removed _num_samples, num_train_samples, num_test_samples_ as argument until [issue 1683](https://github.com/LLNL/lbann/issues/1683) is fixed, data subselection can be accomplished with _percent_to_use_ flag.

**Comment by timmoon10** [comment]
This doesn't pass Bamboo, even with #1685. It seems to be mostly errors when importing submodules in the Python front-ends. It will take a bit more finagling to get working.

**Comment by timmoon10** [comment]
I've debugged the Python errors. It turns out the problem is because PR #1538 broke support for specifying layer datatypes in the Python front-end.

**Pull request merged**